tests/test_evpn_l3_vxlan.py

```diff
@@ -0,0 +1,1726 @@
+
+        attr_keys = {entry[0] for entry in fvs}
+        assert attr_keys == set(expected_attributes.keys())
+
+        for name, value in fvs:
+            assert expected_attributes[name] == value, "Wrong value %s for the attribute %s = %s" % \
+                                  (value, name, expected_attributes[name])
+
+
+class TestL3Vxlan(object):
+
+    def get_vxlan_obj(self):
+        return VxlanTunnel()
+
+    def setup_db(self, dvs):
+        self.pdb = swsscommon.DBConnector(0, dvs.redis_sock, 0)
+        self.adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+        self.cdb = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+
+#    Test 1 - Create and Delete SIP Tunnel and VRF VNI Map entries
+#    @pytest.mark.skip(reason="Starting Route Orch, VRF Orch to be merged")
+    @pytest.mark.dev_sanity
+    def test_sip_tunnel_vrf_vni_map(self, dvs, testlog):
+        vxlan_obj = self.get_vxlan_obj()
+
+        self.setup_db(dvs)
+        tunnel_name = 'tunnel_2'
+        map_name = 'map_1000_100'
+        vrf_map_name = 'evpn_map_1000_Vrf-RED'
+
+        vxlan_obj.fetch_exist_entries(dvs)
+
+        print "\n\nTesting Create and Delete SIP Tunnel and VRF VNI Map entries"
+        print "\tCreate SIP Tunnel"
+        create_vlan1(dvs,"Vlan100")
+        create_vxlan_tunnel(dvs, tunnel_name, '6.6.6.6')
+        create_evpn_nvo(dvs, 'nvo1', tunnel_name)
+        print "\tPass"
+
+        print "\tCreate Vlan-VNI map and VRF-VNI map"
+        create_vxlan_tunnel_map(dvs, tunnel_name, map_name, '1000', 'Vlan100')
+
+        vxlan_obj.create_vrf(dvs, "Vrf-RED")
+        create_vxlan_vrf_tunnel_map(dvs, 'Vrf-RED', '1000')
+        print "\tPass"
+
+        #vlanlist = ['100', '101', '102']
+        #vnilist = ['1000', '1001', '1002']
+
+        print "\tTesting VRF-VNI map in APP DB"
+        vlanlist = ['100']
+        vnilist = ['1000']
+
+        exp_attrs = [
+                ("vni", "1000"),
+        ]
+        exp_attr = {}
+        for an in xrange(len(exp_attrs)):
+            exp_attr[exp_attrs[an][0]] = exp_attrs[an][1]
+
+        check_object(self.pdb, "VRF_TABLE", 'Vrf-RED', exp_attr)
+
+        exp_attrs1 = [
+                ("vni", "1000"),
+                ("vlan", "Vlan100"),
+        ]
+        exp_attr1 = {}
+        for an in xrange(len(exp_attrs1)):
+            exp_attr1[exp_attrs1[an][0]] = exp_attrs1[an][1]
+
+        check_object(self.pdb, "VXLAN_VRF_TABLE", "%s:%s" % (tunnel_name, vrf_map_name), exp_attr1)
+        print "\tPass"
+
+        print "\tTesting SIP Tunnel Creation"
+        vxlan_obj.check_vxlan_sip_tunnel(dvs, tunnel_name, '6.6.6.6', vlanlist, vnilist)
+        print "\tPass"
+
+        print "\tTesting Tunnel Vlan VNI Map Entry"
+        vxlan_obj.check_vxlan_tunnel_vlan_map_entry(dvs, tunnel_name, vlanlist, vnilist)
+        print "\tPass"
+
+        print "\tTesting Tunnel VRF VNI Map Entry"
+        vxlan_obj.check_vxlan_tunnel_vrf_map_entry(dvs, tunnel_name, 'Vrf-RED', '1000')
```

Comment:
Please have the VRF Name defined in a variable and use it everywhere. So that, it might not needed to be changed, in lot of places, if we use a different VRF other than Vrf-RED.
And the same goes with VLAN as well.

Reply:
ok. Will do.